src/components/confirmation_dialog.tsx

```diff
@@ -0,0 +1,48 @@
+import React from "react";
+import { Button, Modal } from "@patternfly/react-core";
+import { useDialogs } from "dialogs";
+
+import cockpit from "cockpit";
+
+const _ = cockpit.gettext;
+
+export const ConfirmationDialog = ({
+    title,
+    callback,
+    content,
+}: {
+  title: string;
+  callback: () => void;
+  content?: React.ReactNode;
+}) => {
+    const Dialogs = useDialogs();
+
+    return (
+        <Modal
+      title={title}
+      variant="small"
+      onClose={Dialogs.close}
+      isOpen
+      footer={
+          <>
+              <Button
+            variant="danger"
+            onClick={() => { callback(); Dialogs.close() }}
+            aria-label={title}
+              >
+                  {_("Delete")}
+              </Button>
+              <Button
+            variant="link"
+            className="btn-cancel"
+            onClick={Dialogs.close}
+              >
+                  {_("Cancel")}
+              </Button>
+          </>
+      }
+        >
+            {content}
```

Comment:
It's a small thing but the content should probably called `children` and follow the standard react pattern
https://react.dev/learn/passing-props-to-a-component#passing-jsx-as-children